### PR TITLE
feat: support user-facing version

### PR DESCRIPTION
**Pull request by Xunzhuo**
Resolves: https://github.com/envoyproxy/gateway/issues/498

Fixes: https://github.com/envoyproxy/gateway/issues/615

Requires: https://github.com/envoyproxy/gateway/pull/639

image

Signed-off-by: bitliu <bitliu@tencent.com>

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -17,32 +17,22 @@
...
+gatewayAPIVersion = os.environ["GATEWAYAPI_VERSION"]
 
 project = f'Envoy Gateway {version}'
 author = 'Envoy Gateway Project Authors'
 
-copyright = '2022 Envoy Gateway Project Authors | ' + fullversion
-
-envoyVersion = os.environ["ENVOY_VERSION"]
-gatewayAPIVersion = os.environ["GATEWAYAPI_VERSION"]
+copyright = '2022 Envoy Gateway Project Authors | ' + version
```
Comment (danehans) [inline docs/conf.py:27]: Let's drop the year from the copyright.

xref: https://github.com/envoyproxy/gateway/pull/590/files

File: README.md [hunk 2]
```diff
@@ -10,8 +10,8 @@ Kubernetes-based application gateway.
 
 * [Blog][blog] introducing Envoy Gateway.
 * [Goals](GOALS.md)
-* [Quickstart](./docs/user/QUICKSTART.md) to use Envoy Gateway in a few simple steps.
-* [Roadmap](./docs/design/ROADMAP.md)
+* [Quickstart](./docs/user/quickstart.md) to use Envoy Gateway in a few simple steps.
+* [Roadmap](./docs/design/roadmap.md)
```
Comment (danehans) [inline README.md:14]: These have already been updated: https://raw.githubusercontent.com/envoyproxy/gateway/main/README.md

**Review by danehans — CHANGES_REQUESTED** [review]
Looks like you have stale quickstart-related changes that should be removed from this PR.

**Comment by danehans** [comment]
IMHO having too many options can cause confusion. I prefer `envoy-gateway version` and the output is the EG, Gateway API, Envoy versions and commit SHA.

**Comment by AliceProxy** [comment]
> 

Plus one on keeping it simple. For example, you can `go version` but not also `go --version` so there is an established pattern for only supporting one style.

**Comment by Xunzhuo** [comment]
Yes I am intending to just keep `eg version`.

**Comment by Xunzhuo** [comment]
I have removed the --version for syncing the style.

**Comment by arkodg** [comment]
> IMHO having too many options can cause confusion. I prefer `envoy-gateway version` and the output is the EG, Gateway API, Envoy versions, and the commit SHA.
> 
> Thoughts @arkodg @youngnick @skriss @LukeShu @AliceProxy

+ 1 on a single pattern, either `envoy-gateway version` or `envoy-gateway --version`
dont have a strong opinion on either

**Review by Xunzhuo — COMMENTED** [review]
/wait 

I am not sure and not recomment to put this into v0.2.0, I think a couple of things need more discussion.

**Comment by danehans** [comment]
> /wait
I am not sure and not recomment to put this into v0.2.0, I think a couple of things need more discussion.

@Xunzhuo converted to draft in the meantime.

**Review by danehans — APPROVED** [review]

**Pull request merged**